note.ts

```diff
@@ -0,0 +1,11 @@
+export interface Note {
+	type: "success" | "warning" | "danger" | "primary"
+	message: string
+}
+export namespace Note {
+	export function is(value: any | Note): value is Note {
+		return typeof(value) == "object" &&
+			(value.type == ("success" || "warning" || "danger" || "primary")) &&
```

Comment:
I'm certain that this does not work the way you intended. It will be true for any type that is truthy.